client/client.go

```diff
@@ -717,13 +740,101 @@ func newWebsocketDialerTLS(conf Config) (*websocket.Dialer, error) {
 
 	dialer := websocket.Dialer{
 		NetDialTLSContext: func(_ context.Context, network string, addr string) (net.Conn, error) {
-			return dialOpenSSL(ctx, &conf, network, addr)
+			proxyURL, err := ProxyURLFromHostPortGetter(addr)
+			if err != nil {
+				return nil, errors.Wrapf(err,
+					"Failed to get http-proxy configurations during websocket dial")
+			}
+			return dialOpenSSL(ctx, &conf, network, addr, proxyURL)
 		},
 	}
 
 	return &dialer, nil
 }
 
+var ProxyURLFromHostPortGetter = func(addr string) (*url.URL, error) {
```

Comment:
As this is such an usual method to be public, please add a minimal documentation for it.

Reply:
Added! :)